src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/FormPass.php

```diff
@@ -59,10 +59,21 @@ public function process(ContainerBuilder $container)
                 throw new \InvalidArgumentException(sprintf('Tagged form type extension must have the extended type configured using the extended_type/extended-type attribute, none was configured for the "%s" service.', $serviceId));
             }
 
-            $typeExtensions[$extendedType][] = $serviceId;
+            $typeExtensions[$extendedType][] = [$serviceId, isset($tag[0]['priority']) ? $tag[0]['priority'] : 0];
         }
 
-        $definition->replaceArgument(2, $typeExtensions);
+        $sortedTypeExtensions = [];
+        foreach ($typeExtensions as $extendedType => $extensionsWithPriority) {
+            usort($extensionsWithPriority, function (array $a, array $b) {
```

Comment:
You can use the [trait](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/DependencyInjection/TaggedContainerInterface.php), it will do the job for you :)